scipy/stats/_resampling.py

```diff
@@ -265,10 +269,14 @@ def bootstrap(data, statistic, *, n_resamples=9999, batch=None,
         `statistic`. Memory usage is O(`batch`*``n``), where ``n`` is the
         sample size. Default is ``None``, in which case ``batch = n_resamples``
         (or ``batch = max(n_resamples, n)`` for ``method='BCa'``).
-    vectorized : bool, default: ``True``
+    vectorized : bool, optional
```

Comment:
Oops, I didn't actually change any defaults to `None`. Will do.

Reply:
I was not sure about this. But it makes sense and should be more robust.

Reply:
I think we can manage this without a deprecation warning:

* If the user was passing either `vectorized=True` or `vectorized=False` explicitly, nothing changes.
* If the user was relying on the `bootstrap` default `vectorized=True`, then (if their code is working) their `statistic` must already have `axis` and be vectorized.
* If the user was relying on the `monte_carlo_test` or `permutation_test` default `vectorized=False`, then nothing changes unless their `statistic` 1) has a keyword argument called `axis` and 2) is not actually vectorized. I wouldn't expect many cases of this, and rather than silently producing the wrong result, they'll get an error because the results of the statistic will not be the right shape (unless they've done something _really_ wonky that I'm having trouble predicting). We'll also put this in the release notes so that users can watch out.

Reply:
Yes agreed it should be transparent for users because we expected `axis`.